otx/cli/manager/config_manager.py

```diff
@@ -231,14 +232,21 @@ def auto_split_data(self, data_roots: str, task: str):
         dataset = self.dataset_manager.import_dataset(data_root=data_roots, data_format=self.data_format)
         train_dataset = self.dataset_manager.get_train_dataset(dataset)
         val_dataset = self.dataset_manager.get_val_dataset(dataset)
+        test_dataset = self.dataset_manager.get_test_dataset(dataset)
         splitted_dataset = None
         if self.data_format in AUTOSPLIT_SUPPORTED_FORMAT:
             if val_dataset is None:
-                splitted_dataset = self.dataset_manager.auto_split(
-                    task=task,
-                    dataset=train_dataset,
-                    split_ratio=[("train", 0.8), ("val", 0.2)],
-                )
+                if self.data_format == "mvtec":
```

Comment:
I agree that the dataset manager would be a more suitable place for this logic. But since auto splitting is performed on the test set for anomaly datasets and on the train set for other datasets, we would have to change the signature of `DatasetManager.auto_split`. Effectively we would have to move almost the entire contents of `ConfigManager.auto_split_data` to `DatasetManager.auto_split`. It's fine by me, but I don't know if this would break the intended design.

Reply:
I'm working on it :)